indico/modules/events/timetable/client/js/dnd/modifiers.ts

```diff
@@ -99,16 +99,28 @@ export function getTotalScroll(element: HTMLElement): {top: number; left: number
 
 /**
  * Restrict the dragged node to be contained within the calendar if it's
- * already scheduled.
+ * already scheduled. Allows pixel limits to be set.
  * @param containerRef React ref to the container element
+ * @param limits Pixel limits on both sides of the y-axis
  * @returns A new Transform object
  */
 export const createRestrictToCalendar =
-  (containerRef): Modifier =>
-  ({draggingNodeRect, transform, id}) => {
-    if (id.startsWith('unscheduled-')) {
-      // return createRestrictToElement(unscheduledRef)({draggingNodeRect, transform, id});
+  (containerRef, limits: [number, number] = [0, 0]): Modifier =>
+  ({id, draggingNodeRect, transform}) => {
+    if (id.startsWith('unscheduled')) {
       return transform;
     }
-    return createRestrictToElement(containerRef)({draggingNodeRect, transform, id});
+
+    let rect = containerRef.current.getBoundingClientRect();
+    const scroll = getTotalScroll(containerRef.current);
+    rect = {
```

Comment:
IMHO this would be clearer than reassigning two somewhat different things to the same variable:

```js
const boundingRect = ...;
const rect = {
  ...
}
```